Src/DependencyCollector/Shared/Implementation/SqlClientDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,539 @@
+        {
+            switch (evnt.Key)
+            {
+                case SqlBeforeExecuteCommand:
+                {
+                    var operationId = (Guid)CommandBefore.OperationId.Fetch(evnt.Value);
+
+                    DependencyCollectorEventSource.Log.SqlClientDiagnosticSubscriberCallbackCalled(operationId, evnt.Key);
+
+                    this.operationStartTimestamps.TryAdd(
+                        operationId,
+                        ConnectionBefore.Timestamp.Fetch(evnt.Value) as long? ?? Stopwatch.GetTimestamp()); // TODO corefx#20748 - timestamp missing from event data
+
+                    break;
+                }
+
+                case SqlAfterExecuteCommand:
+                {
+                    var operationId = (Guid)CommandAfter.OperationId.Fetch(evnt.Value);
+
+                    DependencyCollectorEventSource.Log.SqlClientDiagnosticSubscriberCallbackCalled(operationId, evnt.Key);
+                    
+                    this.OnAfterCommandEvent(
+                        operationId,
+                        (string)CommandAfter.Operation.Fetch(evnt.Value),
+                        (Guid)CommandAfter.ConnectionId.Fetch(evnt.Value),
+                        (SqlCommand)CommandAfter.Command.Fetch(evnt.Value),
+                        (IDictionary)CommandAfter.Statistics.Fetch(evnt.Value),
+                        (long)CommandAfter.Timestamp.Fetch(evnt.Value));
+
+                    break;
+                }
+
+                case SqlErrorExecuteCommand:
+                {
+                    var operationId = (Guid)CommandError.OperationId.Fetch(evnt.Value);
+
+                    DependencyCollectorEventSource.Log.SqlClientDiagnosticSubscriberCallbackCalled(operationId, evnt.Key);
+                    
+                    this.OnCommandErrorEvent(
+                        operationId,
+                        (string)CommandError.Operation.Fetch(evnt.Value),
+                        (Guid)CommandError.ConnectionId.Fetch(evnt.Value),
+                        (SqlCommand)CommandError.Command.Fetch(evnt.Value),
+                        (Exception)CommandError.Exception.Fetch(evnt.Value),
+                        (long)CommandError.Timestamp.Fetch(evnt.Value));
+
+                    break;
+                }
+                    
+                case SqlBeforeOpenConnection:
+                case SqlBeforeCloseConnection:
+                {
+                    var operationId = (Guid)ConnectionBefore.OperationId.Fetch(evnt.Value);
+
+                    DependencyCollectorEventSource.Log.SqlClientDiagnosticSubscriberCallbackCalled(operationId, evnt.Key);
+                    
+                    this.operationStartTimestamps.TryAdd(
+                        operationId,
+                        (long)ConnectionBefore.Timestamp.Fetch(evnt.Value));
+
+                    break;
+                }
+
+                case SqlErrorOpenConnection:
+                case SqlErrorCloseConnection:
```

Comment:
As mentioned in another comment, for consistency with existing module, don't track Close() call.

Reply:
Done.